compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7368,7 +7368,12 @@ private void checkTypeParamExpr(Location pos, BLangExpression arg, BType expecte
             types.checkType(arg.pos, inferredType, expectedType, DiagnosticErrorCode.INCOMPATIBLE_TYPES);
             return;
         }
-        checkExpr(arg, expectedType, data);
+        if (arg.getKind() == NodeKind.REST_ARGS_EXPR && expectedType.tag == TypeTags.UNION) {
```

Comment:
Did you verify, why this is needed?

Reply:
Got a build failure without this. Added `&& expectedType.tag == TypeTags.UNION` as a workaround to avoid the issue with `float:avg(...[1, 2, 3])`. Still checking why this is needed, will update.

Reply:
```
var x = from var {name, price} in [{name: "", price: 2}]
                where name == "XX"
                collect int:max(...[price]);
```
is currently giving `incompatible types: expected '([int,int...]|record {| int n; |})', found '[int...]'` but with this change it won't, and will cause a compiler crash. I think we need to change the approach for this.